src/map/utils/battleutils.cpp

```diff
@@ -1723,14 +1723,10 @@ namespace battleutils
     {
         CItemWeapon* PWeapon = GetEntityWeapon(PDefender, SLOT_MAIN);
 
-        // Defender must have no weapon equipped, or a hand to hand weapon equipped to guard
-        bool validWeapon = (PWeapon == nullptr || PWeapon->getSkillType() == SKILL_HAND_TO_HAND);
+        bool hasGuardSkill = (PDefender->GetMJob() == JOB_MNK || PDefender->GetSJob() == JOB_MNK ||
+                              PDefender->GetMJob() == JOB_PUP || PDefender->GetSJob() == JOB_PUP);
```

Comment:
You don't get skillups from skills that come only from subjobs.

Reply:
He's trying to effect proc rate anyway, not skill gain

Reply:
Sorry, saw "skill rate" and thought this was about rate of gaining skill. Carry on.